### ISSUE TITLE
Unable to find fonts

### ISSUE THREAD

**Issue by bsmithgall** [18F/uswds-rails-gem#7]
Hi!

I am trying to port a rails application over to use this gem (as opposed to a fork of the old [`us_web_design_standards_gem`](https://github.com/18F/us_web_design_standards_gem)), but am getting the following errors after following the instructions:

`Asset filtered out and will not be served: add `Rails.application.config.assets.precompile += %w( sourcesanspro-light-webfont.eot )` to `config/initializers/assets.rb` and restart your server`

I am not the world's foremost Rails developer so it is very possible that this is entirely my fault.

---

My `app/assets/stylesheets/application.scss`:

``` css
@import 'uswds/all';
@import 'uswds_rails_overrides/all';
/* the rest of the file */
```

My `app/views/layouts/application.html.erb`:

``` html
  <!-- 128x128 -->
  <link rel="shortcut icon" type="image/ico" href="<%= asset_path "favicons/favicon.ico" %>" />
  <link rel="icon" type="image/png" href="<%= asset_path "favicons/favicon.png" %>" />

  <!-- 192x192, as recommended for Android
  http://updates.html5rocks.com/2014/11/Support-for-theme-color-in-Chrome-39-for-Android
  -->
  <link rel="icon" type="image/png" sizes="192x192" href="<%= asset_path "favicon-192.png" %>" />

  <!-- 57x57 (precomposed) for iPhone 3GS, pre-2011 iPod Touch and older Android devices -->
  <link rel="apple-touch-icon-precomposed" href="<%= asset_path "favicon-57.png" %>">
  <!-- 72x72 (precomposed) for 1st generation iPad, iPad 2 and iPad mini -->
  <link rel="apple-touch-icon-precomposed" sizes="72x72" href="<%= asset_path "favicon-72.png" %>">
  <!-- 114x114 (precomposed) for iPhone 4, 4S, 5 and post-2011 iPod Touch -->
  <link rel="apple-touch-icon-precomposed" sizes="114x114" href="<%= asset_path "favicon-114.png" %>">
  <!-- 144x144 (precomposed) for iPad 3rd and 4th generation -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="<%= asset_path "favicon-144.png" %>">

  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
```

**Comment by pkarman** [comment]
Thanks for the bug report @bsmithgall -- this gem is very likely still buggy since we've just started it. We'll take a look to try and reproduce this error.

**Comment by jessieay** [comment]
This error makes sense. I adapted the updates for rails font stuff from https://github.com/18F/micropurchase, where I removed all of our USWDS scss but in that project we do have the fonts in `app/assets` there. 

Can do a bit more digging on that one if it is helpful to you, @pkarman . 

In the meantime, @bsmithgall, if you want to get up and running you can download the fonts you need from https://github.com/18F/uswds-rails-gem/tree/master/vendor/assets/fonts and put them in `app/assets/fonts` and that should work. Thanks for reporting this! 

Very helpful.

**Comment by jessieay** [comment]
Ok, I just updated a branch to use the uswds-rails gem and removed all assets and fonts. I am not seeing this error locally when I serve up my app or when I compile assets.

@bsmithgall can you paste the contents of your `config/initializers/assets.rb` file? Also, at what point did you see this error? (usually assets are not precompiled locally)

**Comment by jessieay** [comment]
Here is my diff for anyone who is interested: https://github.com/18F/micropurchase/compare/jy-uswds-gem

**Comment by jessieay** [comment]
Figured it out! I was not seeing this issue locally but when I ran a cucumber test with javascript I saw a similar error:

```
No route matches [GET] "/fonts/sourcesanspro-light-webfont.woff" (ActionController::RoutingError)
```

I didn't see it locally, so realized it must be related to precompiling assets. When I precompiled assets locally, I looked into my compiled application.css file and realized that we are referencing fonts in `vendor` at an incorrect location. For example, one font is ref'd at `'../fonts/sourcesanspro-light-webfont',` but this is SCSS inside `vendor/assets/stylesheets/uswds/core/_fonts.scss` and the font file is definitely not one dir up because we added the `uswds` subdir.

My guess on why it fails during precompilation is that even though we override this font declaration, the asset pipeline is still trying to interpret that SCSS and gets a 404.

While we _could_ update the path to be correct, that might be problematic from a maintenance perspective since we are trying to override stuff in `vendor` not edit it.

Ideas from anyone on how we can fix this? Confirmed that commenting out the SCSS in `vendor/assets/stylesheets/uswds/core/_fonts.scss` fixes this problem.

**Comment by pkarman** [comment]
Maybe we could work with the USWDS folks to make the relative path `../fonts/....` use a variable instead? `$font_path/...`  that way there could be a default value of `../fonts` but we could override?

**Comment by pkarman** [comment]
e.g. http://stackoverflow.com/questions/17089717/how-to-overwrite-scss-variables-when-compiling-to-css

**Comment by bsmithgall** [comment]
@jessieay I don't know if it's still valuable but I haven't actually touched my `assets.rb` file:

``` ruby
# Be sure to restart your server when you modify this file.

# Version of your assets, change this if you want to expire all your assets.
Rails.application.config.assets.version = '1.0'

# Add additional assets to the asset load path
# Rails.application.config.assets.paths << Emoji.images_path

# Precompile additional assets.
# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
# Rails.application.config.assets.precompile += %w( search.js )
```

**Comment by jessieay** [comment]
@pkarman great idea, I will submit a PR to them.

Only challenge with that is it will be a little bit of time before it gets merged and released, in the meantime this is broken.

Maybe short term solution is to comment out code in `vendor/assets/stylesheets/uswds/core/_fonts.scss` ?

**Comment by jessieay** [comment]
Submitted a PR to web design standards: https://github.com/18F/web-design-standards/pull/1407

**Comment by pkarman** [comment]
@bsmithgall can you check whether current master HEAD fixes your problem? The PR mentioned here was merged and we should now be using `$fonts-path` in our overrides.

**Comment by jessieay** [comment]
I don't think we are using the override yet, but if you set it manually it should work 😁

**Comment by jessieay** [comment]
Hello @bsmithgall ! We just merged a fix for this via #12 , please run `bundle update uswds-rails` and try it out (and let us know if it works, if you can!)

Note that you will need to add the following to the top of `application.scss`:

```
@import 'uswds_rails_overrides/core/defaults';
```

This file includes the namespaced `$font-path` scss var and tells the font-face mixin that we are using the rails asset pipeline. 

So there are now 3 lines total required in `application.scss` to use the uswds-rails gem. Hoping to streamline in the future but that appears to be working when I test for now!

**Comment by bsmithgall** [comment]
This worked as expected! Thanks.
